vnext/ReactUWP/Views/KeyboardEventHandler.cpp

```diff
@@ -0,0 +1,405 @@
+  {winrt::VirtualKey::Escape, "Escape"},
+  {winrt::VirtualKey::Execute, "Execute"},
+  {winrt::VirtualKey::Help, "Help"},
+  {winrt::VirtualKey::Pause, "Pause"},
+  {winrt::VirtualKey::Select, "Select"},
+
+  //Device keys
+  {winrt::VirtualKey::Snapshot, "PrintScreen"},
+  {winrt::VirtualKey::Sleep, "Standby"},
+
+  //Common IME keys
+  {winrt::VirtualKey::Convert, "Convert"},
+  {winrt::VirtualKey::Final, "FinalMode"},
+  {winrt::VirtualKey::ModeChange, "ModeChange"},
+  {winrt::VirtualKey::NonConvert, "NonConvert"},
+
+  //Korean keyboards only
+  {winrt::VirtualKey::Hangul, "HangulMode"},
+  {winrt::VirtualKey::Hanja, "HanjaMode"},
+  {winrt::VirtualKey::Junja, "JunjaMode"},
+
+  //Japanese keyboards only
+  {winrt::VirtualKey::Kana, "KanaMode"},
+  {winrt::VirtualKey::Kanji, "KanjiMode"},
+
+  //Function keys
+  {winrt::VirtualKey::F1, "F1"},
+  {winrt::VirtualKey::F2, "F2"},
+  {winrt::VirtualKey::F3, "F3"},
+  {winrt::VirtualKey::F4, "F4"},
+  {winrt::VirtualKey::F5, "F5"},
+  {winrt::VirtualKey::F6, "F6"},
+  {winrt::VirtualKey::F7, "F7"},
+  {winrt::VirtualKey::F8, "F8"},
+  {winrt::VirtualKey::F9, "F9"},
+  {winrt::VirtualKey::F10, "F10"},
+  {winrt::VirtualKey::F11, "F11"},
+  {winrt::VirtualKey::F12, "F12"},
+  {winrt::VirtualKey::F13, "F13"},
+  {winrt::VirtualKey::F14, "F14"},
+  {winrt::VirtualKey::F15, "F15"},
+  {winrt::VirtualKey::F16, "F16"},
+  {winrt::VirtualKey::F17, "F17"},
+  {winrt::VirtualKey::F18, "F18"},
+  {winrt::VirtualKey::F19, "F19"},
+  {winrt::VirtualKey::F20, "F20"},
+
+  //Numeric keypad keys
+  {winrt::VirtualKey::Decimal, "Decimal"},
+  {winrt::VirtualKey::Multiply, "Multiply"},
+  {winrt::VirtualKey::Add, "Add"},
+  {winrt::VirtualKey::Divide, "Divide"},
+  {winrt::VirtualKey::Subtract, "Subtract"},
+  {winrt::VirtualKey::Separator, "Separator"},
+
+  {winrt::VirtualKey::Number0, "0"},
+  {winrt::VirtualKey::Number1, "1"},
+  {winrt::VirtualKey::Number2, "2"},
+  {winrt::VirtualKey::Number3, "3"},
+  {winrt::VirtualKey::Number4, "4"},
+  {winrt::VirtualKey::Number5, "5"},
+  {winrt::VirtualKey::Number6, "6"},
+  {winrt::VirtualKey::Number7, "7"},
+  {winrt::VirtualKey::Number8, "8"},
+  {winrt::VirtualKey::Number9, "9"},
+  //
+};
+
+string KeyboardHelper::FromVirtualKey(winrt::VirtualKey virtualKey, bool shiftDown, bool capLocked)
+{
+  char key = static_cast<char>(virtualKey);
+
+  if (!isalnum(key))
+  {
+    auto it = g_virtualKeyToString.find(virtualKey);
+    if (it == g_virtualKeyToString.end())
+      return "Unidentified";
+    else
+      return it->second;
+  }
+
+  // https://docs.microsoft.com/en-us/uwp/api/windows.system.virtualkey
+  // Virtual Keys for 0-9 and A-Z, they're just aligned to their ASCII representation (in uppercase, for the alphabet VKs)
+  // Windows doesn't define key for a-z, convert it lower if shift is down.
+  if (isalpha(key) && ((!shiftDown && !capLocked) || (shiftDown && capLocked)))
+      key = static_cast<char>(tolower(key));
```

Comment:
This isn't necessary as our API is oriented around virtual keys.  It could also cause confusion for apps as they should be expecting virtual key - let's use the (upper-case) virtual key code.

Reply:
Done